Modules/IO/NIFTI/test/itkNiftiLargeImageRegionReadTest.cxx

```diff
@@ -0,0 +1,68 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkTestingMacros.h"
+
+
+int
+itkNiftiLargeImageRegionReadTest(int argc, char * argv[])
+{
+
+  if (argc != 2)
+  {
+    std::cerr << "Missing arguments" << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv) << " outputFileName" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+
+  constexpr int Dimension = 3;
+  using PixelType = unsigned short;
+
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  // Create a large image
+  ImageType::SizeType   size = { { 1034, 1034, 1020 } };
+  ImageType::RegionType region;
+  region.SetSize(size);
+
+  auto image = ImageType::New();
+  image->SetRegions(region);
+  image->Allocate(true);
+
+  const std::string fname{ argv[1] };
+  itk::WriteImage(image, fname);
+
+
+  using ReaderType = itk::ImageFileReader<ImageType>;
+  auto reader = ReaderType::New();
+
+  reader->SetFileName(fname);
+
+  // Ensure that a sufficiently large subregion of an abritrary pixel type image can be read
+  region.SetSize({ { 1029, 1029, 1017 } });
+  reader->GetOutput()->SetRequestedRegion(region);
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(reader->Update());
```

Comment:
Is there anything more you could possibly check? Only just checking that a function call throws no exception seems a bit minimal to me. I mean, an `Update()` implementation that just _does nothing at all_ would then also pass the test.

Reply:
OK, thanks. I still find it interesting that even an Update() function that does _nothing at all_ would pass the test, but no problem.

Reply:
I'm happy to take suggestions on what to check considering the problems pointed in the issue. Otherwise, I'm fine with the test as it is.

Reply:
Thanks Jon, I was thinking of something like the following, _after_ the Update() call:

    const auto output = reader->GetOutput();
    ITK_TEST_EXPECT_TRUE(output != nullptr);
    ITK_TEST_EXPECT_TRUE(output->GetBufferedRegion() == region);

Does that look reasonable to you as well? To check that the Update() has indeed generated an image of the requested size.

Reply:
Sounds reasonable. Done. Force-pushed. Thanks Niels.